src/Arcus.Messaging.Health/Tcp/TcpHealthServer.cs

```diff
@@ -0,0 +1,161 @@
+            {
+                return Task.FromCanceled(cancellationToken);
+            }
+
+            try
+            {
+                _logger.LogTrace("Starting TCP server...");
+                _listener.Start();
+                _logger.LogInformation("TCP server started!");
+
+                return base.StartAsync(cancellationToken);
+            }
+            catch (Exception exception)
+            {
+                return Task.FromException(exception);
+            }
+        }
+
+        /// <summary>
+        /// This method is called when the <see cref="T:Microsoft.Extensions.Hosting.IHostedService" /> starts. The implementation should return a task that represents
+        /// the lifetime of the long running operation(s) being performed.
+        /// </summary>
+        /// <param name="stoppingToken">Triggered when <see cref="M:Microsoft.Extensions.Hosting.IHostedService.StopAsync(System.Threading.CancellationToken)" /> is called.</param>
+        /// <returns>A <see cref="T:System.Threading.Tasks.Task" /> that represents the long running operations.</returns>
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            while (!stoppingToken.IsCancellationRequested)
+            {
+                _logger.LogInformation("Worker running at: {time}", DateTimeOffset.Now);
+
+                await AcceptConnectionAsync();
+            }
+        }
+
+        private async Task AcceptConnectionAsync()
+        {
+            _logger.LogTrace("Accepting TCP client...");
+            using (TcpClient client = await _listener.AcceptTcpClientAsync())
+            {
+                _logger.LogInformation("TCP client accepted!");
+                using (NetworkStream ns = client.GetStream())
+                {
+                    // TODO: something to serialize as BSON instead?
+                    HealthReport report = await _healthService.CheckHealthAsync();
+                    string serialized = JsonConvert.SerializeObject(report, _serializerSettings);
+
+                    byte[] response = Encoding.UTF8.GetBytes(serialized);
+                    ns.Write(response, 0, response.Length);
+                    _logger.LogInformation("Return health report: {status}", report.Status);
+
+                    ns.Close();
+                    client.Close();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Triggered when the application host is performing a graceful shutdown.
+        /// </summary>
+        /// <param name="cancellationToken">Indicates that the shutdown process should no longer be graceful.</param>
+        public override Task StopAsync(CancellationToken cancellationToken)
+        {
+            if (cancellationToken.IsCancellationRequested)
+            {
+                return Task.FromCanceled(cancellationToken);
+            }
+
+            try
+            {
+                _logger.LogTrace("Stopping TCP server...");
```

Comment:
Again, port

Reply:
Main reason is that you don't always have that context here if you browse the logs